Documentation/sources/content-card-ui/public-classes/contentcarduieventlistening.md

```diff
@@ -0,0 +1,57 @@
+# ContentCardUIEventListening
+
+A protocol that defines methods for listening to UI events related to content cards.
+
+## Protocol Definition
+
+```swift
+public protocol ContentCardUIEventListening {
+    func onDisplay(_ card: ContentCardUI)
+    func onDismiss(_ card: ContentCardUI)
+    func onInteract(_ card: ContentCardUI, _ interactionId: String, actionURL: URL?) -> Bool
+}
```

Comment:
optional keyword can only be marked on @Objc protocols... It will be syntactically wrong to have it marked optional in the protocol definition. 

The swift way declaring it optional is by providing default implementation..  Which we have done in our SDK for ContentCardUIEventListening